### ISSUE TITLE
Bug: helm lint|template causes unexpected error "CA secret is not present in the release namespace"

### ISSUE THREAD

**Issue by Kradepon** [cockroachdb/helm-charts#394]
Chart version: 13.0.0
helm version: v3.9.1

My `values.yaml`
```yaml annotate
cockroachdb:
  image:
    repository: cockroachdb/cockroach
    tag: v24.1.0
  conf:
    port: 26257
    cache: 25%
    max-sql-memory: 25%
  tls:
    enabled: true
    certs:
      selfSigner:
        enabled: true
        caProvided: true
        caSecret: cockroachdb-ca
```

In the charts [helm template](https://github.com/cockroachdb/helm-charts/blob/58c2e86e49b380d763d4e2cb58d729796af4592e/cockroachdb/templates/_helpers.tpl#L200)

```helm annotate
{{/*
Validate that if caProvided is true, then the caSecret must not be empty and secret must be present in the namespace.
*/}}
{{- define "cockroachdb.tls.certs.selfSigner.caProvidedValidation" -}}
{{- if .Values.tls.certs.selfSigner.caProvided -}}
{{- if eq "" .Values.tls.certs.selfSigner.caSecret -}}
    {{ fail "CA secret can't be empty if caProvided is set to true" }}
{{- else -}}
    {{- if not (lookup "v1" "Secret" .Release.Namespace .Values.tls.certs.selfSigner.caSecret) }}
        {{ fail "CA secret is not present in the release namespace" }}
    {{- end }}
{{- end -}}
{{- end -}}
{{- end -}}
```

the line `if not (lookup "v1" "Secret" .Release.Namespace .Values.tls.certs.selfSigner.caSecret)` causes any attempt to `helm lint` or `helm template` to fail with the appropriate error, although the secret is present in the release namespace. Is there any way to skip this validation or how is this supposed to work?

**Comment by Kradepon** [comment]
Puh, leaving this comment for future developers searching for an answer:

There is a long discussion on the [helm community](https://github.com/helm/helm/issues/8137) on the behavior of `lookup` in any `lint`, `template` or `install --dry-run` invocation.

Long story short: running `helm install --dry-run=server release .` will do the proper lookup and work as expected. This option is implemented since [helm 3.13](https://github.com/helm/helm/releases/tag/v3.13.0)

**Comment by rsaelim** [comment]
I am deploying crdb using argo + helm, so not running `helm lint|template` per se, but I'm seeing this error as well and I do have the secret name (configured in values.yaml) in the namespace in which I am attempting to install crdb in, but I'm seeing this error as well from the validation. Any clue?
